packages/chains/src/chains.ts

```diff
@@ -0,0 +1,20 @@
+export {
+  mainnet,
+  goerli,
+  bsc,
+  bscTestnet,
+  zkSync,
+  zkSyncTestnet,
+  // opBNB,
+  opBNBTestnet,
+  polygonZkEvm,
+  polygonZkEvmTestnet,
+  arbitrum,
+  arbitrumGoerli,
+  scrollSepolia,
+  scrollTestnet,
+  linea,
+  lineaTestnet,
+  base,
+  baseGoerli,
+} from 'viem/chains'
```

Comment:
is there a reason for this export? If we need them somewhere else we can simply import from viem?

Reply:
in my view, this pkg is the single source of truth for pancakeswap chains, so all the chain define should in this pkg, include the predefine chain(directly reexport from viem), and global custom chain

Reply:
we could define our own typed chain object later on if that's the case, and we don't have to follow the naming convention and type definition from viem.

Let's discuss the things we need for the chain type after setting up the pkg and for now let's not re-export from viem

Reply:
any examples of our chain object type? as far as I discovered, it used like the same as viem
https://github.com/pancakeswap/pancake-frontend/blob/c899962c860f6e9d4aaf90ff66822a21613911fa/apis/farms/src/provider.ts#L17-L96

Reply:
we may extend on top of that or we can choose to have our own type. For example we may add subgraph links to the config

Reply:
and probably we don't even need to export the viem related chain config as we may provide the viem client out of the box, e.g.

```typescript
import { viemProviders }  from '@pancakeswap/chains'
// or maybe
import { getViemClient }  from '@pancakeswap/chains`

const client = getViemClient({ chainId })
```

so far the most common usecase for the viem chain configs are creating public clients. We can wrap it up so we don't need to define the providers all over the place

Reply:
and our config could be like something else
```typescript
const ethereum = {
  rpcUrls: [],
  subgraphs: {
    v3: '',
    v2: '',
    block: '',
  },
  ...
}
```

of course if we need to export our extended viem chain configs we can surely do, although I think no need for now
```typescript
import { mainnet as defaultMainnet } from 'viem/chains'

const mainnet = {
  ...defaultMainnet,
  // extended by our own config maybe
  rpcUrls: {
    // read from our own rpc url configs
  },
}

export const viemChains = {
  mainnet
}
```

Reply:
we can leave this for further discussion for now. Let's push out the 0.0.1 version only including the `ChainId` first